autorest/codegen/templates/config.py.jinja2

```diff
@@ -20,7 +20,18 @@ def __init__(
 # coding=utf-8
 {{ code_model.options['license_header'] }}
 
-{{ imports }}
+{{ imports.non_typing() }}
+{% if imports.has_typing %}
+
+try:
+    from typing import TYPE_CHECKING
```

Comment:
I would prefer this to be part of the import_serializer.py instead of duplicating it everywhere